File: src/main/java/de/rubixdev/enchantedshulkers/EnchantableBlockEntity.java
package de.rubixdev.enchantedshulkers;

import net.minecraft.nbt.NbtCompound;
import net.minecraft.nbt.NbtList;

public interface EnchantableBlockEntity {
    NbtList getEnchantments();

    void setEnchantments(NbtList enchantments);

    default NbtCompound toClientNbt() {
        // Only send the enchantments to the client to reduce packet size
        NbtCompound nbt = new NbtCompound();
        nbt.put("Enchantments", getEnchantments());
        return nbt;
    }
}

File: versions/26.2/src/main/kotlin/de/rubixdev/enchantedshulkers/interfaces/EnchantableBlockEntity.kt
package de.rubixdev.enchantedshulkers.interfaces

import net.minecraft.world.item.enchantment.ItemEnchantments

/**
 * 26.2 port. Was NbtList-based (`enchantedShulkers$getEnchantments(): NbtList`); block entities
 * still use NBT for their own persisted state in 26.2 (only ItemStack lost its NBT API - see the
 * port report), but [ItemEnchantments] is the natural Mojmap type to carry this data now (it's
 * what `EnchantmentHelper`/the enchantment component system consumes directly, avoiding a
 * NbtList<->ItemEnchantments conversion at every use site), and it has its own [ItemEnchantments.CODEC]
 * that works directly with the new `ValueInput`/`ValueOutput` block-entity persistence API (see
 * ChestBlockEntityMixin).
 */
@Suppress("FunctionName")
interface EnchantableBlockEntity {
    fun `enchantedShulkers$getEnchantments`(): ItemEnchantments

    fun `enchantedShulkers$setEnchantments`(enchantments: ItemEnchantments)
}
